components/ILIAS/Badge/classes/class.ilBadgePersonalTableGUI.php

```diff
@@ -118,9 +126,106 @@ public function getRows(
             $records = \array_slice($records, $range->getStart(), $range->getLength());
         }
 
+        $parent_obj_ids = [];
+        $identifications = [];
+
         foreach ($records as $record) {
-            yield $row_builder->buildDataRow((string) $record['id'], $record);
+            $badge = $record['badge'];
+            $parent_obj_ids[] = $badge->getParentId();
+            if ($badge->getImageRid()) {
+                $identifications[] = $badge->getImageRid();
+            }
+        }
+        $this->irss->preload($identifications);
+        $this->object_data_cache->preloadObjectCache(array_unique($parent_obj_ids));
+
+        foreach ($records as $record) {
+            yield $row_builder->buildDataRow((string) $record['id'], $this->enrichRecord(
+                $record,
+                $access_cache,
+                $ref_id_cache
+            ));
+        }
+    }
+
+    /**
+     * @param array<int, bool> $access_cache
+     * @param array<int, int> $ref_id_cache
+     * @param array{
+     *     id: int,
+     *     title_sortable: string,
+     *     awarded_by_sortable: string,
+     *     badge_issued_on: DateTimeImmutable,
+     *     active: bool,
+     *     assignment: ilBadgeAssignment,
+     *     badge: ilBadge
+     * } $record
+     * @return array
+     */
+    private function enrichRecord(array $record, array &$access_cache, array &$ref_id_cache): array
+    {
+        $badge = $record['badge'];
+        $ass = $record['assignment'];
+
+        $parent = null;
+        if ($badge->getParentId()) {
+            $parent = $badge->getParentMeta();
+            if ($parent['type'] === 'bdga') {
+                $parent = null;
+            }
+        }
```

Comment:
IMO this `getParentMeta` result can be cached as well. Furthermore, this is also used in `getRows` to determine a sortable parent title, so we could also fill/use the cache there.